src/Utilities/StdHelpers.hpp

```diff
@@ -389,3 +391,16 @@ inline std::array<T, Dim> operator-(const std::array<T, Dim>& rhs) noexcept(
   }
   return result;
 }
+
+template <size_t Dim, typename T>
+inline T magnitude(const std::array<T, Dim>& rhs) noexcept(
```

Comment:
It looks like as written this is only useful for T=double (or float or complex, but we don't use those currently), due to default initialization being assumed to give zero and the requirement that T be passable to std::sqrt.  We should either make it work for DataVector or remove the template parameter.